http/src/request/mod.rs

```diff
@@ -69,10 +68,10 @@ pub(crate) fn audit_header(
     reason: &str,
 ) -> Result<impl Iterator<Item = (HeaderName, HeaderValue)>, Error> {
     let header_name = HeaderName::from_static("x-audit-log-reason");
-    let encoded_reason = utf8_percent_encode(reason, NON_ALPHANUMERIC).to_string();
-    let header_value = HeaderValue::from_str(&encoded_reason).map_err(|e| Error {
+    let encoded_reason = urlencoding::encode(reason);
+    let header_value = encoded_reason.parse().map_err(|e| Error {
         kind: ErrorType::CreatingHeader {
-            name: encoded_reason.clone(),
+            name: encoded_reason.into_owned(),
```

Comment:
This always allocates with `percent_encoding`, `urlencoding` doesn't allocate if `reason` is already correctly encoded (most likely a single word).

One of the reasons I opened this PR was because I misread [https://lib.rs](https://lib.rs)'s stats and thought [urlencoding](https://lib.rs/crates/urlencoding) was more popular than [percent-encoding](https://lib.rs/crates/percent-encoding), as that's not the case I'll close this PR.